akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -358,19 +362,10 @@ object Source {
    * If the [[Future]] fails the stream is failed with the exception from the future. If downstream cancels before the
    * stream completes the materialized `Future` will be failed with a [[StreamDetachedException]]
    */
+  @deprecated("Use 'Source.futureSource' (potentially together with `Source.fromGraph`) instead", "2.6.0")
   def fromFutureSource[T, M](future: Future[Graph[SourceShape[T], M]]): Source[T, Future[M]] =
     fromGraph(new FutureFlattenSource(future))
 
-  /**
-   * Streams the elements of an asynchronous source once its given `completion` operator completes.
-   * If the [[CompletionStage]] fails the stream is failed with the exception from the future.
-   * If downstream cancels before the stream completes the materialized `Future` will be failed
-   * with a [[StreamDetachedException]]
-   */
-  def fromSourceCompletionStage[T, M](
-      completion: CompletionStage[_ <: Graph[SourceShape[T], M]]): Source[T, CompletionStage[M]] =
-    fromFutureSource(completion.toScala).mapMaterializedValue(_.toJava)
```

Comment:
removed without deprecation?

Reply:
Ah, damned, I thought I added it but was sure it belonged in the Scala API. I'll re-add